### ISSUE TITLE
Spacemacs stopped working

### ISSUE THREAD

**Issue by oiwn** [syl20bnr/spacemacs#5687]
I got some strange errors after i tried to update packages. So i decided to completely reinstall spacemacs.

I did 

```
git clone https://github.com/syl20bnr/spacemacs ~/.emacs.d
```

deleted my .spacemacs file.

And got this...

```
Warning (spacemacs): An error occurred while retrieving the theme, using default theme. (error: (error Package ‘spacemacs-theme-’ is unavailable))
Warning (initialization): An error occurred while loading ‘/Users/..../.emacs.d/init.el’:

error: Package ‘evil-’ is unavailable

To ensure normal operation, you should investigate and remove the
cause of the error in your initialization file.  Start Emacs with
the ‘--debug-init’ option to view a complete error backtrace.
```

[Screenshot1](http://i.imgur.com/aXl6G8B.png)

[Screenshot2](http://i.imgur.com/uNbtvyq.png)

**Comment by syl20bnr** [comment]
Try with this:
- wipe `elpa` directory
- start emacs with `emacs --insecure`

**Comment by istinspring** [comment]
yea it works for "clean install" but after i add any layer it show me errors.

```
An error occurred while installing company-tern (error: (cl-assertion-failed (or (= (buffer-size tar-data-buffer) (buffer-size)) (eq tar-data-swapped (> (buffer-size tar-data-buffer) (buffer-size))))))
```

**Comment by syl20bnr** [comment]
Try with the variable `dotspacemacs-elpa-https`set to `nil`.

**Comment by istinspring** [comment]
elpa-https same as --insecure

spacemacs clean installation works with this flag, but can't add any additional packages. 

```
error: Package 'package-build-' is unavailable
```
something is wrong.

> actually this is a javascript layer error. This is out of the scope for this issue.

it was just an example, all additional packages like this. I changed internet connection to mobile. Same problems.

**Comment by istinspring** [comment]
well looks like it solved with dotspacemacs-elpa-https 

But why? Just few days ago everything was fine.

**Comment by StreakyCobra** [comment]
Have you changed your location? Changed your internet service provider? Change your network configuration? etc…?

**Comment by istinspring** [comment]
yea i was on trip for few days in another country. But i tried few connections already. Now i'm using mobile but without "elpa-https nil" won't work anyway.

**Comment by StreakyCobra** [comment]
In which country do you live? Some people reported having problem in China because of the GFW. Otherwise I don't know what could be the cause.

**Comment by istinspring** [comment]
In Thailand now.

**Comment by CodeFalling** [comment]
What dose `emacs --insecure` do? I use this and problem gone. I am in China.

**Comment by StreakyCobra** [comment]
We now use `https` for retrieving packages from MELPA, and it seems that this may be problematic in China for some people. The `--insecure` flag force to use `http`.

**Comment by geolibres** [comment]
PROBLEM IS :    http://spacemacs.org/doc/FAQ  In 3.3 Why are all packages unavailable?
DIAGNOSTIG  is : Go to scratch or (gnutls-available-p)  
or: M x enter (gnutls-available-p) enter
if n : 
I SOLVE  it by following the instructions  https://www.gnu.org/software/emacs/manual/html_mono/emacs-gnutls.html#Help-For-Users   

 
(Metakey=alt-key emacs)

**Comment by jgomo3** [comment]
For me the problem was caused by slow connection. Improved it and the installation from scratch worked.

I imporoved it by pausing Dropbox. Not possible to improve it for everybody.

**Comment by slahn** [comment]
Had this same problem when installing on a windows computer - it was because GnuTLS was not bundled with the windows version of Emacs. Solved it by installing GnuTLS.

Spacemacs could probably 'solve' this by checking `gnutls-available-p` before it tries to install over https, and show a better warning message. But maybe this is not a high priority :)

**Comment by danieloosthuizen** [comment]
I've got the same issue, and it started when I updated to using Emacs 25. (Using `yaourt emacs25-git`.)

I've since removed `emacs25-git` and have removed and reinstalled Emacs 24.5 using `pacman`, but the problem persists. Even the `--insecure` flag doesn't seem to help a fresh install of Spacemacs; neither does changing `dotspacemacs-elpa-https`.

Though it's possible my problems are for different reasons, they show the same symptoms, so I'm putting this here.

**Comment by malcook** [comment]
on OSX, had same problem with fresh emacs 25.1 installed via https://emacsformacosx.com/
problem persisted even after `homebrew install gnutls`
problem persisted even after installing emacs fresh with homebrew
however, the problem went away with a fresh install of spacemacs AFTER installing gnutls

speculation: reinstall spacemacs after ensuring that gnutls is available (as reported by  (gnutls-available-p) )

**Comment by danieloosthuizen** [comment]
*OK!* I finally snapped, and had a good hard look at the `*Backtrace*`.

The problem looks to be with `package-archives` – `configuration-layer/initialize` *should* set this to include MELPA, etc., **unless** `package--initialized` is `t`. I've `ag`ed through `.spacemacs.d`, and the only places I've found `package--initialized` is [here](https://github.com/syl20bnr/spacemacs/blob/master/core/core-configuration-layer.el#L277) and [here](https://github.com/syl20bnr/spacemacs/blob/master/layers/+spacemacs/spacemacs-ui/packages.el#L248).

**Comment by danieloosthuizen** [comment]
Turns out it's nothing to do with that!

The way I had my stuff setup was to have a `.emacs.d` and a `.spacemacs.d`, and then have the following in my `.emacs.d`:

```
(setq use-spacemacs t)   ; or nil 

(when use-spacemacs
  (setq user-emacs-directory "~/.spacemacs.d/"))   ; defaults to ~/.emacs.d/

(load (expand-file-name "init.el" user-emacs-directory))
```

This breaks with Emacs 25, so I've gotten rid of the old `.emacs.d` and renamed my `.spacemacs.d`.

Thanks!

**Comment by argentum47** [comment]
Facing the same problem here. (india). On fedora, connection times out within seconds. But in normal emacs list-packages works fine. Tried with --insecure too. Few days ago I installed on nother machine. was fine

**Comment by andialbrecht** [comment]
I've had the same issue. For me it worked to set `dotspacemacs-elpa-https` to `nil` and increase the timeout: `dotspacemacs-elpa-timeout 30`. My connection is most of the time pretty ok, but it seems that on a fresh install some initial request takes longer than expected. Just a guess...

**Comment by netroby** [comment]
The same issue. seems emacs can not download package via http proxy ?

**Comment by BranislavLazic** [comment]
@andialbrecht Proposal worked for me. I increased timeout to 30. Thanks.

**Comment by peterchealse** [comment]
--insecure works

**Comment by jollyrogue** [comment]
I'm getting the same error on a brand new FreeBSD install, and `emacs --insecure` fixes the problem. Adding Gnu TLS helps somewhat, but it doesn't fix the entire problem. Setting `dotspacemacs-elpa-timeout 30` helps a little as well, but not entirely. Setting `dotspacemacs-elpa-https nil` fixes eveything.

Details:
- FreeBSD 12.0-release-p9
- emacs-nox-26.2,3 installed via pkg
- gnutls-3.6.8 installed via pkg

Debug output attached.

[spacemacs-debug.log](https://github.com/syl20bnr/spacemacs/files/3489274/spacemacs-debug.log)

**Comment by duianto** [comment]
@jollyrogue 
It's probably related to this issue:
Failed to download 'gnu' archive. Bad request. (Solution included) #12535

The linked Emacs bugreport says that the fix will be included in the next version:
Found in version 26.1
Fixed in versions 27.1, 26.3
https://debbugs.gnu.org/cgi/bugreport.cgi?bug=34341
